internal/service/common/device.go

```diff
@@ -138,6 +139,19 @@ func updateServerSideDeviceStatus(device *api.Device) bool {
 	return device.Status.Summary.Status != lastDeviceStatus
 }
 
+func updateServerSideLifecycleStatus(device *api.Device) bool {
+	lastLifecycleStatus := device.Status.Lifecycle.Status
+
+	if device.IsDecommissioning() {
+		device.Status.Lifecycle = api.DeviceLifecycleStatus{
+			Info:   util.StrToPtr("Device has acknowledged decommissioning request"),
+			Status: api.DeviceLifecycleStatusDecommissioning,
+		}
+	}
+
+	return device.Status.Lifecycle.Status != lastLifecycleStatus
```

Comment:
If we are not decommissioning, isn't this always false? This logic works, so this is just an observation.

Reply:
Basically yes; the lifecycle under normal circumstances can transition only from Enrolled -> Decommissioning or from Decommissioning -> Decommissioned. There is also a `LifecycleStatusUnknown` value but there is no scenario setting it to that currently.